### PR TITLE
chore(makefile): create check makefile target

### PR DESCRIPTION
**Pull request by jack-w-shaw**
## Description

The check target is a special target used in our CI.

This should allow our default Jenkins jobs to correctly run our unit tests and merge

## Type of change

- Change in tests (one or several tests have been changed)
- Maintenance work (repository related, like Github actions, or revving the Go version, etc.)

## QA steps

commenting `/build` or `/merge` correctly trigger jenkins jobs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,5 +101,4 @@ install-snap-dependencies: check-snap-package
 install-dependencies: install-snap-dependencies
 ## install-dependencies: Install all the dependencies
 
-
-
+check: juju-unit-test
```
Comment (jack-w-shaw) [inline Makefile:104]: @hmlanigan `juju-unit-test` seems to be quite a lightweight test suite.

Shall we put something else here? Or is this good for now? What do you think?

Comment (jack-w-shaw) [reply Makefile:104]: Imo best option should be to keep it light in Jenkins, and make more github actions required

Comment (hmlanigan) [reply Makefile:104]: This set of tests will get bigger over time, but okay for now.

**Review by hmlanigan — APPROVED** [review]

**Comment by hmlanigan** [comment]
Need a signed commit to merge, and unblock other PRs which now need the check make target.

**Pull request merged**